test/e2e/networkpolicy_test.go

```diff
@@ -270,6 +271,165 @@ func TestDefaultDenyEgressPolicy(t *testing.T) {
 	}
 }
 
+// TestEgressToServerInCIDRBlock is a duplicate case as "should allow egress access to server in CIDR block [Feature:NetworkPolicy]",
+// while a bug exists in TOT release. This should be deleted when upstream is updated.
+// https://github.com/kubernetes/kubernetes/blob/v1.20.0-alpha.0/test/e2e/network/network_policy.go#L1365
+func TestEgressToServerInCIDRBlock(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	workerNode := workerNodeName(1)
+	serverAName, serverAIPs, cleanupFunc := createAndWaitForPod(t, data, data.createNginxPodOnNode, "test-server-", workerNode)
+	defer cleanupFunc()
+	serverBName, serverBIPs, cleanupFunc := createAndWaitForPod(t, data, data.createNginxPodOnNode, "test-server-", workerNode)
+	defer cleanupFunc()
+
+	clientA, _, cleanupFunc := createAndWaitForPod(t, data, data.createBusyboxPodOnNode, "test-client-", workerNode)
+	defer cleanupFunc()
+	var serverCIDR string
+	var serverAIP, serverBIP string
+	if serverAIPs.ipv6 != nil {
+		serverCIDR = fmt.Sprintf("%s/128", serverAIPs.ipv6.String())
+		serverAIP = serverAIPs.ipv6.String()
+		serverBIP = serverBIPs.ipv6.String()
+	} else {
+		serverCIDR = fmt.Sprintf("%s/32", serverAIPs.ipv4.String())
+		serverAIP = serverAIPs.ipv4.String()
+		serverBIP = serverBIPs.ipv4.String()
+	}
+
+	if err := data.runNetcatCommandFromTestPod(clientA, serverAIP, 80); err != nil {
+		t.Fatalf("%s should be able to netcat %s", clientA, serverAName)
+	}
+	if err := data.runNetcatCommandFromTestPod(clientA, serverBIP, 80); err != nil {
+		t.Fatalf("%s should be able to netcat %s", clientA, serverBName)
+	}
+
+	np, err := data.createNetworkPolicy("allow-client-a-via-cidr-egress-rule", &networkingv1.NetworkPolicySpec{
+		PodSelector: metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"antrea-e2e": clientA,
+			},
+		},
+		PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeEgress},
+		Egress: []networkingv1.NetworkPolicyEgressRule{
+			{
+				To: []networkingv1.NetworkPolicyPeer{
+					{
+						IPBlock: &networkingv1.IPBlock{
+							CIDR: serverCIDR,
+						},
+					},
+				},
+			},
+		},
+	})
+	if err != nil {
+		t.Fatalf("Error when creating network policy: %v", err)
+	}
+	cleanupNP := func() {
+		if np == nil {
```

Comment:
if we get to that point, I think `np` is guaranteed not to be `nil`, otherwise `createNetworkPolicy` would have returned an error.

maybe this can be simplified to:
```go
defer func() {
    if err := data.deleteNetworkpolicy(np); err != nil {
        t.Errorf("Error when deleting network policy: %v", err)
    }
}
```

also not sure why you are explicitly setting `np` to `nil` at the end of the function

Reply:
Yes... Actually I copied the code block from another Netpol testcase and I thought such defer func was expected.
https://github.com/vmware-tanzu/antrea/blob/master/test/e2e/networkpolicy_test.go#L292